File: TrainingCshar/Formulaio/FApiTask.cs
using CsharLibrary.Class.Data_Process;
using CsharLibrary.Models;
using System;
using System.Text.RegularExpressions;
using System.Web.Script.Serialization;
using System.Windows.Forms;

namespace TrainingCshar.Formulaio
{
    public partial class FApiTask : Form
    {
        private IManagement management;
        private readonly ToolTip tlp = new ToolTip();

        public FApiTask(IManagement _management)
        {
            InitializeComponent();
            management = _management;
        }

        private void LbResponse_Click(object sender, EventArgs e)
        {
            tlp.Show("Elige un metodo de envio en la lista", this.lbResponse, 30);
        }

        private void CmbTypeEnv_SelectedIndexChanged(object sender, EventArgs e)
        {
            bool estado = cmbTypeEnv.SelectedIndex > 0 ? true : false;
            txtUrl.Enabled = estado;
            btnSend.Visible = estado;
        }

        private void TxtUrl_TextChanged(object sender, EventArgs e)
        {
            Regex reg = new Regex(management.GetPattern(), RegexOptions.Compiled | RegexOptions.IgnoreCase);

            bool isUri = reg.IsMatch(txtUrl.Text);
            btnSend.Enabled = isUri;
            chkTexto.Visible = isUri;
        }

        private void ChkTexto_CheckedChanged(object sender, EventArgs e)
        {
            txtJson.Enabled = chkTexto.Checked;
            txtJson.ReadOnly = !chkTexto.Checked;
            btnToJson.Visible = chkTexto.Checked;
            btnToClass.Visible = chkTexto.Checked;
        }

        private void TxtJson_TextChanged(object sender, EventArgs e)
        {
            try
            {
                lbStatus.ResetText();
                JavaScriptSerializer javaScript;

                javaScript = new JavaScriptSerializer();

                Person objectPersona = javaScript.Deserialize<Person>(txtJson.Text);
                var jsonPersona = javaScript.Serialize(txtJson.Text);
            }
            catch (Exception except)
            {
                lbStatus.Text = except.Message.Substring(0, 34);
            }
        }

        private async void btnSend_ClickAsync(object sender, EventArgs e)
        {
            txtJson.Text = await management.GetHttpUrl(txtUrl.Text);
        }
    }
}

File: TrainingCshar/Formulaio/FApiTask.vb
Imports System
Imports System.Text.RegularExpressions
Imports System.Windows.Forms
Imports System.Web.Script.Serialization
Imports ClassVbExtendss

Namespace TrainingCshar.Formulaio
    Public Partial Class FApiTask
        Inherits Form

        Private tlp As ToolTip = New ToolTip()

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub LbResponse_Click(ByVal sender As Object, ByVal e As EventArgs)
            tlp.Show("Elige un metodo de envio en la lista", lbResponse, 30)
        End Sub

        Private Sub CmbTypeEnv_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs)
            Dim estado = If(cmbTypeEnv.SelectedIndex > 0, True, False)
            txtUrl.Enabled = estado
            btnSend.Visible = estado
        End Sub

        Private Sub TxtUrl_TextChanged(ByVal sender As Object, ByVal e As EventArgs)
            Dim pattern = "^(http|https|ftp|)\://|[a-zA-Z0-9\-\.]+\.[a-zA-Z](:[a-zA-Z0-9]*)?/?([a-zA-Z0-9\-\._\?\,\'/\\\+&amp;%\$#\=~])*[^\.\,\)\(\s]$"
            Dim reg As Regex = New Regex(pattern, RegexOptions.Compiled Or RegexOptions.IgnoreCase)
            Dim isUri = reg.IsMatch(txtUrl.Text)
            btnSend.Enabled = isUri
            chkTexto.Visible = isUri
        End Sub

        Private Sub ChkTexto_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs)
            txtJson.Enabled = chkTexto.Checked
            txtJson.ReadOnly = Not chkTexto.Checked
            btnToJson.Visible = chkTexto.Checked
            btnToClass.Visible = chkTexto.Checked
        End Sub

        Private Sub TxtJson_TextChanged(ByVal sender As Object, ByVal e As EventArgs)
            Try
                lbStatus.ResetText()
                Dim javaScript As JavaScriptSerializer
                javaScript = New JavaScriptSerializer()
                Dim objectPersona = javaScript.Deserialize(Of Persona)(txtJson.Text)
                Dim jsonPersona = javaScript.Serialize(txtJson.Text)
            Catch except As Exception
                lbStatus.Text = except.Message.Substring(0, 34)
            End Try
        End Sub
    End Class
End Namespace
